types/versionlockstring.pp

```diff
@@ -40,4 +44,6 @@
 # @example An invalid string (wildcard spans the VERSION and RELEASE fields): `0:bash-4.*-el6.x86_64
 # @example An invlaid string (wildcard spans the VERSION, RELEASE, and ARCH fields): `0:bash-*`
 #
+# lint:ignore:140chars
 type Yum::VersionlockString = Pattern[/^([0-9\*]+):([0-9a-zA-Z\._\+%\{\}\*-]+)-([^-]+)-([^-]+)\.(([0-9a-zZ-Z_\*]+)(?:\.(noarch|x86_64|i386|arm|ppc64|ppc64le|sparc64|ia64|alpha|ip|m68k|mips|mipsel|mk68k|mint|ppc|rs6000|s390|s390x|sh|sparc|xtensa|\*))?)$/]
```

Comment:
wouldn't it be nice if puppet's Pattern new about [`/x`](https://ruby-doc.org/core-2.7.1/Regexp.html#class-Regexp-label-Free-Spacing+Mode+and+Comments) (and also about %{}… but that's another dream)